### ISSUE TITLE
Trait support

### ISSUE THREAD

**Issue by suhr** [rust-analyzer/rust-analyzer#909]
> Failed to exactly resolve the symbol. This is probably because rust_analyzer does not yet support glob imports or traits.

**Comment by kjeremy** [comment]
This is intentional. Can you give an example of where you are seeing this where you believe that it should work?

**Comment by suhr** [comment]
Even if it's intentional, it would be really nice to have a tracking issue, so people could know when this is implemented.

**Comment by DJMcNab** [comment]
We do support glob imports now I believe! Trait support however is quite far in the future.

**Comment by colelawrence** [comment]
What is the next step to moving this forward? Is this waiting on something else?

**Comment by colelawrence** [comment]
Related: "Chalk RLS integration" https://paper.dropbox.com/doc/Chalk-RLS-integration--AaBRVmOY2Qv~GNc~8re8bL_RAg-QCJelXDeaq7GV6jVQln4j

**Comment by CptFox** [comment]
@DJMcNab Any roadmap on trait support ? The document on Chalk RLS Integration is a bit opaque to me, any parts where some help is needed to enable trait support ? :)

**Comment by flodiebold** [comment]
@CptFox The foundations are there, and basic trait method completions are working. They're not working in all cases yet though, because a) there are still a bunch of missing features and b) there are some performance improvements needed.

I'm working on support for associated types right now, I'm making progress but haven't had that much time to invest in the last few weeks. It's hard to mentor further improvements to the Chalk integration before that's done, though.

**Comment by flodiebold** [comment]
There are still some missing edge cases, and associated type projection often doesn't work because of a Chalk issue, but I think this issue has outlived its usefulness :slightly_smiling_face:
